test/time_offset_test.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* time_t/offset (+/-XXXX) tests for ASN1 and X509 */
+
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+
+#include <openssl/asn1.h>
+#include <openssl/x509.h>
+#include "testutil.h"
+#include "test_main.h"
+#include "e_os.h"
+
+typedef struct {
+    const char *data;
+    int time_result;
+    int type;
+} TESTDATA;
+
+
+/**********************************************************************
+ *
+ * Test driver
+ *
+ ***/
+
+static TESTDATA tests[] = {
+    { "20001201000000Z",      0, V_ASN1_GENERALIZEDTIME },
+    { "20001201010000+0100",  0, V_ASN1_GENERALIZEDTIME },
+    { "20001201050000+0500",  0, V_ASN1_GENERALIZEDTIME },
+    { "20001130230000-0100",  0, V_ASN1_GENERALIZEDTIME },
+    { "20001130190000-0500",  0, V_ASN1_GENERALIZEDTIME },
+    { "20001130190001-0500",  1, V_ASN1_GENERALIZEDTIME }, /* +1 second */
+    { "20001130185959-0500", -1, V_ASN1_GENERALIZEDTIME }, /* -1 second */
+    { "001201000000Z",        0, V_ASN1_UTCTIME },
+    { "001201010000+0100",    0, V_ASN1_UTCTIME },
+    { "001201050000+0500",    0, V_ASN1_UTCTIME },
+    { "001130230000-0100",    0, V_ASN1_UTCTIME },
+    { "001130190000-0500",    0, V_ASN1_UTCTIME },
+    { "001201000000-0000",    0, V_ASN1_UTCTIME },
+    { "001201000001-0000",    1, V_ASN1_UTCTIME }, /* +1 second */
+    { "001130235959-0000",   -1, V_ASN1_UTCTIME }, /* -1 second */
+    { "20001201000000+0000",  0, V_ASN1_GENERALIZEDTIME },
+    { "20001201000000+0100", -1, V_ASN1_GENERALIZEDTIME },
+    { "001201000000+0100",   -1, V_ASN1_UTCTIME },
+    { "20001201000000-0100",  1, V_ASN1_GENERALIZEDTIME },
+    { "001201000000-0100",    1, V_ASN1_UTCTIME },
```

Comment:
This is specifically testing offsets, so not sure what you are looking for? Invalid time formats do not really fall under umbrella of this particular change (I would even argue out of scope).

I have something more generic for invalid times under PR #2753